index.js

```diff
@@ -1,35 +1,23 @@
 // Public intefrace
 module.exports = promisedPipe
 
+const ensureFunctionType = (fn, i) => {
+    if (typeof fn !== 'function') {
+        throw Error(`pipe requires each argument to be a function. Argument #${i+1} is of type "${typeof fn}"`)
+    }
+}
+
+const chain = (q, fn) => q.then(fn)
+
 function promisedPipe(...fns) {
     if (fns.length < 1) {
         throw Error('pipe requires at least one argument')
     }
 
-    fns.forEach((fn, i) => {
-        if (typeof fn !== 'function') {
-            throw Error(`pipe requires each argument to be a function. Argument #${i+1} is of type "${typeof fn}"`)
-        }
-    })
+    fns.forEach(ensureFunctionType)
 
```

Comment:
it's an optimization, since we don't need anything form the scope, there is no value in creating this function every time we call it.
It's minor, but I think it also makes it clear that the function doesn't need anything from it's parent scope